### PR TITLE
#1941 DateUtils test fix

### LINKED ISSUE #1941: DateUtilsTest. getWeekdaysInMonth[jvm] test failing
## Pre-submission checklist
- [X] I am submitting a bug report, not a feature request.
- [X] I have searched for similar issues, but did not find any matches.

## Description
During the execution of `DateUtilsTest. getWeekdaysInMonth[jvm]`, the test fails. The rough description of the test is:
```
java.lang.AssertionError: 
Expected: [<4>, <4>, <4>, <4>, <5>, <5>, <5>]
     but: was [<4>, <4>, <4>, <4>, <4>, <4>, <4>]
...
```

I checked this test, and indeed, Febraury 2018 has different set of weekdays (4 mon, 4 tues, 4 weds, 4 fris, 4 sat, 4 suns).
The problem is that `GregorianCalendar` returns Time Zone dependent date, and in my case it was returning a one day before actual date.

## Steps to reproduce
Run `./build.sh build debug` command

## System information
- Any (CI issue)

## Screenshots
Calendar 2018 February:
image

Source of error:
image

And the rest of calendar tests

### PR DESCRIPTION
**Pull request by foralost**
Making GregorianCalendars go UTC to fix TimeZone issues.

Closes #1941.

### CODE REVIEW

File: uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/utils/DateUtilsTest.kt [hunk 1]
```diff
@@ -118,17 +118,24 @@ class DateUtilsTest : BaseUnitTest() {
         assertThat(arrayOf("Sat", "Sun", "Mon", "Tue", "Wed", "Thu", "Fri"), equalTo(longWeekdayNames))
     }
 
+    private fun getCalendarUTC(year: Int, month: Int, dayOfMonth: Int): GregorianCalendar {
```
Comment (hiqua) [inline uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/utils/DateUtilsTest.kt:121]: If we use this only in a single, we might as well put it as a local function inside that test, IMHO.

Comment (foralost) [reply uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/utils/DateUtilsTest.kt:121]: sure

on Sunday I will do it

Comment (foralost) [reply uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/utils/DateUtilsTest.kt]: Force pushed, and tested on APIs 28, 29, 30, everything passes locally.

**Comment by iSoron** [comment]
Thank you for the PR, @foralost. I merged it with some minor changes.

**Pull request merged**